### PR TITLE
Compability/promises

### PR DESCRIPTION
**Pull request by spotscale**
This fixes most of the Warnings from bluebird when using Promise.defered. I had trouble fixing the last defered in readPrimitiveSet due to the nature of that function. If some one could resolve that I would be happy.

### CODE REVIEW

**Comment by spotscale** [comment]
Any comments on this?

File: tests/osgDB/Input.js [hunk 1]
```diff
@@ -203,13 +203,15 @@ module.exports = function () {
...
         } );
-        input.readPrimitiveSet().then( function ( /*value */) {
+        input.readPrimitiveSet().then( function ( /* value */) {
             return input.setJSON( {
                 'DrawArrays': {
                     'UniqueID': 10
                 }
             } ).readPrimitiveSet();
         } ).then( function ( o2 ) {
+            console.log( '>>>' );
```
Comment (cedricpinson) [inline tests/osgDB/Input.js]: I guess it was for debugging, could you remove it ?

Comment (cedricpinson) [inline tests/osgDB/Input.js]: You could probably squash all commits into one. Also keep commits message clear and not too long like this https://github.com/cedricpinson/osgjs/pull/796/commits/d50f1750b5b1ed4e3a002dd9a6765ee43ad7dee7
If you need to add details or explanation do it after the first line of the commit message

but it's good, just worry about performance

**Comment by spotscale** [comment]
I have done what you asked and merged in master as well. As of performance not writing the warnings to console should save some time. If you notice any performance issues please tell me and I'll fix it.

**Comment by jtorresfabra** [comment]
Hi @spotscale, 

Can you please avoid 'merge' commits, instead of merge you should **rebase** against master.

Thanks!

**Comment by spotscale** [comment]
@jtorresfabra I think I have fixed this now. Sorry.

**Comment by spotscale** [comment]
@cedricpinson How would I run this test?

**Comment by spotscale** [comment]
I think I have solved the performance problem at least I get similar values when testing with the materials example.

File: .idea/markdown-navigator/profiles_settings.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<component name="MarkdownNavigator.ProfileManager">
```
Comment (jtorresfabra) [inline .idea/markdown-navigator/profiles_settings.xml]: You should not add this file to commit

**Comment by cedricpinson** [comment]
what's the state of this PR ?

**Comment by cedricpinson** [comment]
@jtorresfabra any idea ?

**Comment by jtorresfabra** [comment]
I will have a look ASAP

**Comment by jtorresfabra** [comment]
Hey @spotscale could you rebase  + remove .idea/markdown-navigator/profiles_settings.xml ?

**Comment by jtorresfabra** [comment]
Fixed here https://github.com/cedricpinson/osgjs/pull/831

**Pull request closed without merging**